### DISCUSSION TITLE
Why does leaf value in plot tree is related to learning rate?

### DISCUSSION THREAD

**Discussion by ccylance** [catboost/catboost#2634] [Q&A]
catboost version: catboost 1.2
Operating System: macos 14.4.1
CPU: M1
GPU: no

I am trying to understanding the algorithm. According to paper, the leaf value is the average value of the gradients of allocated items in the leaf node. However, in here, I found the leaf value is relate to learning rate? Why was that? And how the leaf value is calculated when using gradient estimate method? It would be better to tell my the function file in the source code
![image](https://github.com/catboost/catboost/assets/54780246/ddf55939-fd0b-4cd9-8c99-fd08c8a7895b)
![image](https://github.com/catboost/catboost/assets/54780246/73cd629c-2c84-48e9-840c-59549c53302c)

**Comment by pabx06** [comment]
Lucky you ? all of my leafs are either 0.00 or -0.00 not relay helpful. seems like hard-coded rounding/truncation.
![image](https://github.com/user-attachments/assets/dafe34fe-fc28-4e84-b364-6ca4f4e465a4)
